src/curve.rs

```diff
@@ -175,19 +205,46 @@ impl<C: Curve> ProjectivePoint<C> {
         let x3 = h * s;
         let y3 = w * (b - h) - rr.double();
         let z3 = s.cube();
-        Self { x: x3, y: y3, z: z3, zero: false }
+        Self {
+            x: x3,
+            y: y3,
+            z: z3,
+            zero: false,
+        }
     }
 
     pub fn add_slices(a: &[Self], b: &[Self]) -> Vec<Self> {
         assert_eq!(a.len(), b.len());
-        a.iter().zip(b.iter()).map(|(&a_i, &b_i)| a_i + b_i).collect()
+        a.iter()
+            .zip(b.iter())
+            .map(|(&a_i, &b_i)| a_i + b_i)
+            .collect()
+    }
+
+    pub fn neg(&self) -> Self {
```

Comment:
There's also an `impl Neg` for AffinePoint & ProjectivePoint, so `-p` should work